### ISSUE TITLE
[Bug]: Author identity unknown | fatal: empty ident name (for <>) not allowed

### ISSUE THREAD

**Issue by nathani-axis** [denolehov/obsidian-git#366]
### Describe the bug

When trying to commit, I get an error.

The same happens when running `git commit -a` outside of obsidian, in a terminal.

The same only happens for the obsidian repo, all other repos get the correct user.name and user.email from the --global git config.

### Relevant errors (if available)

```shell
Uncaught (in promise) Error: Author identity unknown

*** Please tell me who you are.

Run

  git config --global user.email "you@example.com"
  git config --global user.name "Your Name"

to set your account's default identity.
Omit --global to set the identity only in this repository.

fatal: empty ident name (for <>) not allowed

    at Object.action (plugin:obsidian-git:23937:25)
    at PluginStore.exec (plugin:obsidian-git:23962:25)
    at eval (plugin:obsidian-git:21549:43)
    at new Promise (<anonymous>)
    at GitExecutorChain.handleTaskData (plugin:obsidian-git:21547:16)
    at GitExecutorChain.eval (plugin:obsidian-git:21531:44)
    at Generator.next (<anonymous>)
    at fulfilled (plugin:obsidian-git:20380:24)
action @ plugin:obsidian-git:23937
exec @ plugin:obsidian-git:23962
eval @ plugin:obsidian-git:21549
handleTaskData @ plugin:obsidian-git:21547
eval @ plugin:obsidian-git:21531
fulfilled @ plugin:obsidian-git:20380
```

### Steps to reproduce

- have a properly configured git config
- download obsidian-git and set it up
- commit through obsidian-git

### Expected Behavior

when I commit through obsidian-git, it doesn't break my repo somehow.

### Addition context

I have gh (the github cli) installed and configured. I see that obsidian-git in the console is trying to get the username there and gets an error. The obsidian repo however should not need anything from gh.

```
Uncaught (in promise) Error: /usr/bin/gh auth git-credential get: line 1: /usr/bin/gh: No such file or directory
fatal: could not read Username for 'https://github.com': No such device or address

    at Object.action (plugin:obsidian-git:23937:25)
    at PluginStore.exec (plugin:obsidian-git:23962:25)
    at eval (plugin:obsidian-git:21549:43)
    at new Promise (<anonymous>)
    at GitExecutorChain.handleTaskData (plugin:obsidian-git:21547:16)
    at GitExecutorChain.eval (plugin:obsidian-git:21531:44)
    at Generator.next (<anonymous>)
    at fulfilled (plugin:obsidian-git:20380:24)
action @ plugin:obsidian-git:23937
exec @ plugin:obsidian-git:23962
eval @ plugin:obsidian-git:21549
handleTaskData @ plugin:obsidian-git:21547
eval @ plugin:obsidian-git:21531
fulfilled @ plugin:obsidian-git:20380

```

### Operating system

Linux

### Plugin version

2.9.3

**Comment by Vinzent03** [comment]
Can you share your repo .git/config file?

**Comment by nathani-axis** [comment]
I forced the user.email and user.name git config (without the --global) and then it worked, I was able to commit.

**Comment by nathani-axis** [comment]
I had a seperate issue with pushing through a proxy, which made me start tweaking at stuff to make it work.

I removed the name and email (made the fields empty) in the obsidian-git. Could that have wrote empty strings to the git config ?

My assumption was that those fields, when empty, should revert to using --global config instead, if that makes sense. Usually when you leave fields empty, you revert back to the default.

**Comment by Vinzent03** [comment]
Yeah that may be the issue.

**Comment by Metal-Toilet** [comment]
I'm running into a very similar issue. I'm using the same email and password I have on github in git but it won't push. I tried getting rid of --global when setting and had no luck.

**Comment by Vinzent03** [comment]
Setting an empty name or email should be fixed in 2.9.4. This only affects commit and pull. Please create another issue with your exact error message.
